Objects/unicodeobject.c

```diff
@@ -5098,10 +5083,13 @@ find_first_nonascii(const unsigned char *start, const unsigned char *end)
         const unsigned char *p2 = _Py_ALIGN_UP(p, SIZEOF_SIZE_T);
 #if PY_LITTLE_ENDIAN && HAVE_CTZ
         if (p < p2) {
+            size_t u;
 #if defined(_M_AMD64) || defined(_M_IX86) || defined(__x86_64__) || defined(__i386__)
-            size_t u = load_unaligned_x86_amd64(p) & ASCII_CHAR_MASK;
+            // x86 and amd64 are little endian and can load unaligned memory.
+            memcpy(&u, p, sizeof(size_t));
+            u &= ASCII_CHAR_MASK;
```

Comment:
We should use memcpy regardless of the architecture. It doesn't impose any alignment requirements on it's arguments. In other words:

* Get rid of the `#if defined(_M_AMD64) || ...`
* Don't use `load_unaligned`

`memcpy` will generate the correct code even on architectures that do not support unaligned loads: https://gcc.godbolt.org/z/MP7MrKd3b.

Reply:
Is this what you had in mind?